letter-sender/src/test/java/com/christmas/letter/sender/controller/ChristmasLetterControllerTest.java

```diff
@@ -0,0 +1,54 @@
+package com.christmas.letter.sender.controller;
+
+import com.christmas.letter.sender.model.ChristmasLetter;
+import com.christmas.letter.sender.service.ChristmasLetterService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(ChristmasLetterController.class)
+public class ChristmasLetterControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @MockBean
+    private ChristmasLetterService christmasLetterService;
+
+    private static final String PATH = "/letters/send";
+
+    @Test
+    public void whenSendChristmasLetter_withValidRequest_ThenMessageReceivedOnQueue() throws Exception {
+        //Prepare message for the SNS Topic
+        String testMessage = objectMapper.writeValueAsString(new ChristmasLetter("testing@gmail.com", "Luiza", "fimo", "Iasi"));
+
+        //Mock sending the message
+        mockMvc.perform(post(PATH)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(testMessage))
+                .andExpect(status().isCreated());
+    }
+
+    @Test
+    public void whenSendChristmasLetter_withInvalidRequest_ThenReturnBadRequest() throws Exception {
```

Comment:
In a production application, I would expect to have a test for each parameter that is invalid. Also, let's also add checks for the validation details, example "email is required", "email is invalid" etc. If you're not familiar with the concept, and if you are, I would suggest you to look into parameterized tests. 
https://www.baeldung.com/parameterized-tests-junit-5

For me, CsvSource is the simplest method, but you can use whatever you see fit.

Reply:
I parameterized the test, also added extra validation message